config/routes.rb

```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  root "users#index"
+  resources :users do
+    resources :posts
+  end
```

Comment:
- [ ] It would be better to only have the routes you need, instead of having all of the routes (index, show, new, edit, ...etc). You can achieve that by using the following syntax. Exposing unnecessary endpoints in your website can cause bugs and errors that you might not have anticipated.
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
Was actually thinking about using shallow: true. Thanks @ShadyShawkat  for the tip.